### DISCUSSION TITLE
Repocket and other 5 number ports

### DISCUSSION THREAD

**Discussion by Raffy142** [engageub/InternetIncome#160] [Q&A]
Hi, i have problem with proxy 5 ports, when i set i get this error
Can someone help me?

time="2024-10-16T23:26:35Z" level=info msg="[DIALER] set fwmark: 0x22b"
time="2024-10-16T23:26:35Z" level=info msg="[STACK] tun://tun0 <-> http://216.146.20.92:25971"
time="2024-10-16T23:26:37Z" level=info msg="[TCP] 198.18.0.1:46118 <-> 104.21.87.17:443"
time="2024-10-16T23:26:38Z" level=warning msg="[TCP] dial 208.95.112.1:80: HTTP connect status: 403 Forbidden"
time="2024-10-16T23:26:38Z" level=info msg="[TCP] 198.18.0.1:41806 <-> 172.67.139.50:443"
time="2024-10-16T23:26:39Z" level=info msg="[TCP] 198.18.0.1:46130 <-> 104.21.87.17:443"
time="2024-10-16T23:26:39Z" level=warning msg="[TCP] dial 207.246.122.60:7070: HTTP connect status: 403 Forbidden"
time="2024-10-16T23:26:40Z" level=warning msg="[TCP] dial 207.246.122.60:7070: HTTP connect status: 403 Forbidden"
time="2024-10-16T23:26:41Z" level=warning msg="[TCP] dial 207.246.122.60:7070: HTTP connect status: 403 Forbidden"

**Comment by engageub** [comment]
Hello,
Could you please check if these IP's are accessible via your proxy.
You can use curl command as mentioned below. Replace `YOUR_PROXY` with your http proxy. 

```
curl -x YOUR_PROXY http://208.95.112.1
```
Alternatively, you can use browser to access the IP mentioned. 

Thank you

**Reply by Raffy142** [reply]
Yes, this is the result
This problem i have only with repocket and packetshare, honeygain work without any problem

  "status"       : "success",
  "continent"    : "North America",
  "continentCode": "NA",
  "country"      : "United States",
  "countryCode"  : "US",
  "region"       : "NY",
  "regionName"   : "New York",
  "city"         : "Middle Island",
  "district"     : "",
  "zip"          : "11953",
  "lat"          : 40.8858,
  "lon"          : -72.9294,
  "timezone"     : "America/New_York",
  "offset"       : -14400,
  "currency"     : "USD",
  "isp"          : "AT\u0026T Services, Inc.",
  "org"          : "Budget Wireless TNC LLC",
  "as"           : "AS7018 AT\u0026T Services, Inc.",
  "asname"       : "ATT-INTERNET4",
  "mobile"       : true,
  "proxy"        : false,
  "hosting"      : false,
  "query"        : "216.146.20.95"
}

**Reply by engageub** [reply]
May be this IP 208.95.112.1 had problem earlier due to too many requests. Does it happen all the time with this IP address.
Also, does this happen with direct connection?

Thank you

**Reply by Raffy142** [reply]
I just entered the proxies with 5, currently I can tell you that the proxies with 5 ports do not work and those with 4 do

**Reply by engageub** [reply]
Hello,
With reference to 5 ports, do you mean port number `25971` containing 5 digits in `http://216.146.20.92:25971`
From the logs port 443 does not have problem with outgoing connection. 

Thank you

**Reply by Raffy142** [reply]
Yes i mean the 25971, you have some solution? I think the problem is this because on bytelixir i have the same problem and the same error

**Reply by engageub** [reply]
If you see this pattern with other proxy provider also,  please raise an issue with tun2socks.

https://github.com/xjasonlyu/tun2socks/issues


You can also try this with tun2proxy instead of tun2socks to confirm if the issue is with tun2socks.
Please use [tunproxy](https://github.com/engageub/InternetIncome/tree/tunproxy)  branch to use tun2proxy.

Thank you

**Reply by Raffy142** [reply]
This is tun2proxy logs

sudo docker container logs tun5ddc055db8d73638edf67281919a7a811
[2024-10-17T00:02:23Z WARN  tproxy_config::linux] failed to bind mount custom resolv.conf onto /etc/resolv.conf, resorting to direct write
[2024-10-17T00:02:23Z INFO  tun2proxy] Proxy http server: 216.146.20.72:28779
[2024-10-17T00:02:26Z INFO  tun2proxy] Beginning #3 TCP 10.0.0.33:37900 -> 198.18.0.0:443
[2024-10-17T00:02:26Z INFO  tun2proxy] Beginning #7 TCP 10.0.0.33:39916 -> 198.18.0.1:80
[2024-10-17T00:02:27Z INFO  tun2proxy] Beginning #11 TCP 10.0.0.33:37916 -> 198.18.0.0:443
[2024-10-17T00:02:27Z INFO  tun2proxy] Beginning #15 TCP 10.0.0.33:37918 -> 198.18.0.0:443
[2024-10-17T00:02:28Z INFO  tun2proxy] Beginning #16 TCP 10.0.0.33:41980 -> 45.77.146.116:7070
[2024-10-17T00:02:29Z INFO  tun2proxy] Beginning #17 TCP 10.0.0.33:41982 -> 45.77.146.116:7070
[2024-10-17T00:02:30Z INFO  tun2proxy] Beginning #18 TCP 10.0.0.33:36856 -> 45.77.146.116:7070
[2024-10-17T00:02:30Z INFO  tun2proxy] Beginning #19 TCP 10.0.0.33:36868 -> 45.77.146.116:7070
[2024-10-17T00:02:31Z INFO  tun2proxy] Beginning #20 TCP 10.0.0.33:36880 -> 45.77.146.116:7070
[2024-10-17T00:02:32Z INFO  tun2proxy] Beginning #21 TCP 10.0.0.33:36896 -> 45.77.146.116:7070
[2024-10-17T00:02:32Z INFO  tun2proxy] Beginning #22 TCP 10.0.0.33:36904 -> 45.77.146.116:7070
[2024-10-17T00:02:33Z INFO  tun2proxy] Beginning #23 TCP 10.0.0.33:36912 -> 45.77.146.116:7070
[2024-10-17T00:02:34Z INFO  tun2proxy] Beginning #24 TCP 10.0.0.33:36918 -> 45.77.146.116:7070
[2024-10-17T00:02:35Z INFO  tun2proxy] Beginning #25 TCP 10.0.0.33:36922 -> 45.77.146.116:7070
[2024-10-17T00:02:35Z INFO  tun2proxy] Beginning #26 TCP 10.0.0.33:36924 -> 45.77.146.116:7070
[2024-10-17T00:02:36Z ERROR tun2proxy] #2 UDP 10.0.0.33:34809 -> 1.1.1.1:53 error "timed out"
[2024-10-17T00:02:36Z ERROR tun2proxy] #0 UDP 10.0.0.33:34809 -> 8.8.8.8:53 error "timed out"
[2024-10-17T00:02:36Z ERROR tun2proxy] #1 UDP 10.0.0.33:34809 -> 8.8.4.4:53 error "timed out"
[2024-10-17T00:02:36Z INFO  tun2proxy] Beginning #27 TCP 10.0.0.33:36928 -> 45.77.146.116:7070
[2024-10-17T00:02:36Z ERROR tun2proxy] #5 UDP 10.0.0.33:39335 -> 8.8.4.4:53 error "timed out"
[2024-10-17T00:02:36Z ERROR tun2proxy] #6 UDP 10.0.0.33:39335 -> 1.1.1.1:53 error "timed out"
[2024-10-17T00:02:36Z ERROR tun2proxy] #4 UDP 10.0.0.33:39335 -> 8.8.8.8:53 error "timed out"
[2024-10-17T00:02:37Z ERROR tun2proxy] #10 UDP 10.0.0.33:44571 -> 1.1.1.1:53 error "timed out"
[2024-10-17T00:02:37Z ERROR tun2proxy] #8 UDP 10.0.0.33:44571 -> 8.8.8.8:53 error "timed out"
[2024-10-17T00:02:37Z ERROR tun2proxy] #9 UDP 10.0.0.33:44571 -> 8.8.4.4:53 error "timed out"
[2024-10-17T00:02:37Z INFO  tun2proxy] Beginning #28 TCP 10.0.0.33:36930 -> 45.77.146.116:7070
[2024-10-17T00:02:37Z INFO  tun2proxy] Beginning #29 TCP 10.0.0.33:36932 -> 45.77.146.116:7070
[2024-10-17T00:02:37Z ERROR tun2proxy] #14 UDP 10.0.0.33:51711 -> 1.1.1.1:53 error "timed out"
[2024-10-17T00:02:37Z ERROR tun2proxy] #12 UDP 10.0.0.33:51711 -> 8.8.8.8:53 error "timed out"
[2024-10-17T00:02:37Z ERROR tun2proxy] #13 UDP 10.0.0.33:51711 -> 8.8.4.4:53 error "timed out"
[2024-10-17T00:02:38Z INFO  tun2proxy] Beginning #30 TCP 10.0.0.33:36940 -> 45.77.146.116:7070

**Reply by engageub** [reply]
If you have same problem with bytelixir also, either they are using similar code behind or similar firewall or the requests are being blocked by these services itself based on user-agent or IP address. These are the possible reasons, I can think of right now. But you have to try tun2proxy to confirm that the issue is not with tun2socks. 
You can also execute curl command inside the tun2socks container to see the output being displayed. 

Thank you

**Reply by Raffy142** [reply]
I just tried enabling tun2proxy from properties and the problem persists, I can't understand why honeygain works without problems

**Reply by engageub** [reply]
Your tun2proxy logs show that DNS is being timed out.  Something might have been changed in tun2proxy.
Is Honeygain using the same IP address as repocket while making HTTP connections?
Do you see the same IP address 208.95.112.1  being used by honeygain and repocket for getting location info.

**Reply by Raffy142** [reply]
He have the same ip address and the same scheme, copy and paste

**Reply by Raffy142** [reply]
I understood the problem, I tried to configure the IP with socks5 and DNS enabled on properties and now it works, but I am forced to use socks5

**Reply by engageub** [reply]
Yes, HTTP proxies don't support DNS.
You will have to use socks5 proxy with socks5 DNS if you wish to use proxy DNS.
Also, it is not clear if you have the same issue with tun2proxy while using HTTP Proxy with Direct DNS.

Thank you

**Reply by Raffy142** [reply]
I don't know since I haven't tried, although I tried to enable dns with cloudflare from properties.conf and it didn't even work

**Reply by Raffy142** [reply]
One question, for earning is better socks5 than http or is similar?

**Reply by engageub** [reply]
The problem you have based on what is specified is related to TCP not with DNS. So you can use Direct DNS instead of DNS over HTTPS.
Both Socks5 and HTTP support TCP protocols. But UDP protocols are supported only by Socks5 and not HTTP.
Whether SOCKS5 is "better" than HTTP depends on your needs. Here are some key differences:

**Protocol Type**: SOCKS5 is a proxy protocol that works at a lower level, allowing it to handle any type of traffic (TCP and UDP), while HTTP is specifically designed for web traffic.

**Performance**: SOCKS5 can be faster for certain types of data transfers since it doesn't add the overhead associated with HTTP requests and responses.

**Flexibility**: SOCKS5 can handle various protocols and is not limited to web browsing, making it more versatile for different applications (like torrents or games).

**Anonymity**: Both can provide anonymity, but SOCKS5 is generally considered more secure when configured properly, as it can encapsulate traffic more effectively.

**Authentication**: SOCKS5 supports different types of authentication methods, providing better control over who can use the proxy.

**Summary**:
Use an HTTP proxy if you mainly need to browse the web and want features like caching and content filtering.
Use a SOCKS5 proxy if you need to work with multiple types of traffic or require better anonymity and performance for applications beyond just web browsing.

Thank you

**Reply by Raffy142** [reply]
I think the problem is throw the port 80, i think i don’t have access with proxy, bytelixir telling me. U have some solution for bypass this error?

**Reply by engageub** [reply]
Does this happen with socks5 with 5 port numbers?
If bytelixir tells you that, then you have to contact your proxy provider.
Also, get more details from bytelixir how they are sending the request and what parameters are being used.
That would help your proxy provider to understand what is causing the issue.
Since bytelixir supports proxy in dashboard and you don't use any applications. you will have to contact your proxy provider after getting the details from bytelixir.
Please, share the update here after getting the details as it would be useful for others also.


Thank you

**Reply by Raffy142** [reply]
With socks5 in repocket and packetshare device is connected but the earning is very low.
From bytelixir he told me i have problem with 80 port, probably my isp res don’t allow port 80 connection

**Reply by engageub** [reply]
I see that the IP 208.95.112.1 resolves to ip-api.com
Its showing data using curl with http request.
```
curl http://ip-api.com
```
Where as with https it shows 403 forbidden if curl command is used without using any user-agent which by default uses `curl/version` as user-agent. It means that response is also dependent on user-agent being used. 
```
curl -v https://ip-api.com
```
If some random user-agent is used in the http request. it is showing 301 (HTTP redirection) code as its redirecting to https website.
```

curl "random useragent" -v http://ip-api.com
```

If some random user-agent is used in the https request. it is showing 200(HTTP successful) code.
```

curl "random useragent" -v https://ip-api.com
```

Please contact bytelixir how they are sending the request and what parameters are being used. You can then use the same parameters to replicate and inform your proxy provider about it if its causing issues with 5 port HTTP proxy. 

Thank you

**Reply by Raffy142** [reply]
Personally I am interested in solving the problem for repocket and packetshare

**Reply by engageub** [reply]
But with bytelixir you are not using any InternetIncome scripts. So either it is your proxy problem or all the three are using something in common. 
The code may not be opensource to check how the requests are being made. You have to contact them to get the details. 
Have you tried with other proxy provider having 5 ports. 

Thank you

**Reply by Raffy142** [reply]
no but I asked the provider to provide me with other IPs

**Reply by Raffy142** [reply]
I'm trying to use the proxy with windows and navigate, i don't have any problem, is possible?
![image](https://github.com/user-attachments/assets/c7365d5c-eb94-4edc-97b7-0c49073ee22f)

**Reply by engageub** [reply]
In the browser, are you using http or https? http uses port 80.
The user agent may be different in the application when compared to what the browser uses.

**Reply by Raffy142** [reply]
I'm using http

**Reply by engageub** [reply]
I meant to say the browser URL you are using should be http. If it gets redirected to https automatically, its not what we are looking for.
From the logs mentioned above, the errors were related to port 80 where 403 forbidden response was received.
You can try with other proxies where 5 number ports are used.
Please find the list of free proxies using the following URL.

https://proxyscrape.com/free-proxy-list


Thank you

**Reply by Raffy142** [reply]
i try with proxy 5 ports free
this is the result
[2024-10-17T14:49:49Z INFO  tun2proxy] Proxy http server: 136.226.113.91:1006
[2024-10-17T14:50:01Z ERROR tun2proxy] #0 UDP 10.0.0.33:45788 -> 8.8.8.8:53 error "timed out"
[2024-10-17T14:50:01Z ERROR tun2proxy] #1 UDP 10.0.0.33:45788 -> 8.8.4.4:53 error "timed out"
[2024-10-17T14:50:01Z ERROR tun2proxy] #2 UDP 10.0.0.33:45788 -> 1.1.1.1:53 error "timed out"
[2024-10-17T14:50:02Z ERROR tun2proxy] #4 UDP 10.0.0.33:55144 -> 8.8.8.8:53 error "timed out"
[2024-10-17T14:50:02Z ERROR tun2proxy] #6 UDP 10.0.0.33:55144 -> 1.1.1.1:53 error "timed out"
[2024-10-17T14:50:02Z ERROR tun2proxy] #5 UDP 10.0.0.33:55144 -> 8.8.4.4:53 error "timed out"

this is with tun2socks
time="2024-10-17T14:55:48Z" level=warning msg="[TCP] dial 172.67.139.50:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"
time="2024-10-17T14:55:49Z" level=warning msg="[TCP] dial 188.114.96.0:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"
time="2024-10-17T14:55:54Z" level=warning msg="[TCP] dial 172.67.139.50:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"
time="2024-10-17T14:56:00Z" level=warning msg="[TCP] dial 172.67.139.50:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"
time="2024-10-17T14:56:06Z" level=warning msg="[TCP] dial 172.67.139.50:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"
time="2024-10-17T14:56:12Z" level=warning msg="[TCP] dial 172.67.139.50:443: connect to 131.153.187.5:50689: dial tcp 131.153.187.5:50689: i/o timeout"

**Reply by Raffy142** [reply]
what if I use a protocol other than Socks and http? i have IP Authorization too

**Reply by engageub** [reply]
The logs (tun2socks) show timeouts, so those proxies might not be working. Your tun2proxy logs have DNS issues. So there is some problem with your configuration or the latest version of tun2proxy.
Yes, you can use https proxies also but the issue is related to 5 port numbers on HTTP proxies.

Thank you

**Reply by engageub** [reply]
Hello,
I just wanted to follow up on this. Were you able to get proxies with 5 port numbers from other providers or got it working using the free proxies. 

Thank you

**Reply by Raffy142** [reply]
Hi, no, i think the problem is provider and the isp is s***, but with honeygain, pawnapp work without any problem, probably don’t use too much 80 port

**Reply by engageub** [reply]
Thank you for the response. The port 80 is being used to get the IP address of your proxy.
The service providers are using port 80 to send the data when using curl command. So obviously the application will use the same.
It is still not clear what the underlying reason is. 

Thank you
